test/context/snapguidistContextFactory.spec.js

```diff
@@ -0,0 +1,125 @@
+import snapguidistContextFactory from '../../src/context/snapguidistContextFactory'
+
+jest.mock(
+  '../../src/context/testQueueFactory',
+  () => () => {
+    const listeners = new Set()
+    return {
+      addTest: jest.fn(
+        (name, snapshot) => listeners.forEach(ls => ls({ name, response: snapshot }))
+      ),
+      clear: jest.fn(() => listeners.clear()),
+      clearListeners: jest.fn(() => listeners.clear()),
+      clearQueue: jest.fn(),
+      listen: jest.fn((listener) => {
+        listeners.add(listener)
+        return () => listeners.delete(listener)
+      }),
+    }
+  }
+)
+
+jest.mock(
+  '../../src/context/renderSnapshot',
+  () => (name, component) => `{"name":"${name}","tree":"${component}"}`
+)
+
+test('should expose the expected API', () => {
+  const context = snapguidistContextFactory()
+  const { clear, listen, runTest } = context
+
+  expect(clear).toBeDefined()
+  expect(typeof clear).toBe('function')
+  expect(listen).toBeDefined()
+  expect(typeof listen).toBe('function')
+  expect(runTest).toBeDefined()
+  expect(typeof runTest).toBe('function')
+
+  expect(Object.keys(context).length).toBe(3)
+})
+
+test('should add requested snapshot to return true', () => {
+  const { runTest } = snapguidistContextFactory()
+
+  const result = runTest('name', 'reactElement')
+  expect(result).toBeTruthy()
```

Comment:
You're right, also the test is not updated to the actual value returned by `runtTest`, I'll update it